lib/ClangImporter/ImportMacro.cpp

```diff
@@ -424,118 +440,158 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
-    // Check for an expression of the form (FLAG1 | FLAG2), (FLAG1 & FLAG2),
-    // (FLAG1 || FLAG2), or (FLAG1 || FLAG2)
-    } else if (tokenI[0].is(clang::tok::identifier) &&
-               isBinaryOperator(tokenI[1]) &&
-               tokenI[2].is(clang::tok::identifier)) {
-      auto firstID = tokenI[0].getIdentifierInfo();
-      auto secondID = tokenI[2].getIdentifierInfo();
-
-      if (firstID->hasMacroDefinition() && secondID->hasMacroDefinition()) {
-        auto firstMacroInfo = impl.getClangPreprocessor().getMacroInfo(firstID);
-        auto secondMacroInfo = impl.getClangPreprocessor().getMacroInfo(
-                                                                      secondID);
-        auto firstIdentifier =
-            impl.getNameImporter().importMacroName(firstID, firstMacroInfo);
-        auto secondIdentifier =
-            impl.getNameImporter().importMacroName(secondID, secondMacroInfo);
-        impl.importMacro(firstIdentifier, firstMacroInfo);
-        impl.importMacro(secondIdentifier, secondMacroInfo);
-        auto firstIterator = impl.ImportedMacroConstants.find(firstMacroInfo);
-        if (firstIterator == impl.ImportedMacroConstants.end()) {
-          return nullptr;
-        }
-        auto secondIterator = impl.ImportedMacroConstants.find(secondMacroInfo);
-        if (secondIterator == impl.ImportedMacroConstants.end()) {
-          return nullptr;
-        }
-
-        auto firstConstant = firstIterator->second;
-        auto secondConstant = secondIterator->second;
-        auto firstValue = firstConstant.first;
-        auto secondValue = secondConstant.first;
-        if (!firstValue.isInt() || !secondValue.isInt()) {
-          return nullptr;
+                                 ConstantConvertKind::Coerce,
+                                 /*isStatic=*/false, ClangN);
+    }
+    // Subtraction:
+    // (INT1 - INT2)
+    else if (tokenI[1].is(clang::tok::minus)) {
+      llvm::APSInt value = firstValue - secondValue;
+      return createMacroConstant(impl, macro, name, DC, firstType,
+                                 clang::APValue(value),
+                                 ConstantConvertKind::Coerce,
+                                 /*isStatic=*/false, ClangN);
+    }
+    // Left-shift:
+    // (INT1 << INT2)
+    else if (tokenI[1].is(clang::tok::lessless)) {
+      // Left-shift by a negative number is UB in C. Don't import.
+      if (secondValue.isNegative()) { return nullptr; }
+
+      llvm::APSInt value { firstValue.shl(secondValue),
+                           firstValue.isUnsigned() };
+      return createMacroConstant(impl, macro, name, DC, firstType,
+                                 clang::APValue(value),
+                                 ConstantConvertKind::Coerce,
+                                 /*isStatic=*/false, ClangN);
+    }
+    // AND, OR (bitwise, logical):
+    // (INT1 | INT2), (INT1 & INT2), (INT1 || INT2), (INT1 && INT2)
+    else if(isBinaryOperator(tokenI[1])) {
```

Comment:
Nitpick: Missing space, and also we prefer to put `else if` on the same line as the close-brace.